client/src/actions/index.js

```diff
@@ -242,12 +242,14 @@ const requestDifferentialExpression = (set1, set2, num_genes = 10) => async (
     */
     const state = getState();
     const { universe } = state.controls;
-    const set1ByIndex = rangeEncodeIndices(
-      _.map(set1, s => universe.obsNameToIndexMap[s])
-    );
-    const set2ByIndex = rangeEncodeIndices(
-      _.map(set2, s => universe.obsNameToIndexMap[s])
-    );
+
+    // These lines ensure that we convert any TypedArray to an Array.
+    // This is necessary because JSON.stringify() does some very strange
+    // things with TypedArrays (they are marshalled to JSON objects, rather
+    // than being marshalled as a JSON array).
+    const aset1 = Array.isArray(set1) ? set1 : Array.from(set1);
+    const aset2 = Array.isArray(set2) ? set2 : Array.from(set2);
```

Comment:
This was a general fix unrelated to dataframes, right?

Reply:
it is related to the improved "mask" interface to crossfilter.    It now returns a TypedArray rather than an Array,   This in turn was motivated by Dataframe (connecting crossfilter to the Dataframe.icut function)

Reply:
got it